### PR TITLE
feat: added default headers to error headers

### PR DESCRIPTION
**Pull request by bashleigh**
Added default headers to error responses

### CODE REVIEW

File: src/http/http.error.handler.ts [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
+  defaultHeaders: DefaultHeadersInterface = {},
 ): APIGatewayProxyResult => ({
   statusCode: isHttpErrorException(error)
     ? error.getStatus()
     : HttpStatusCode.INTERNAL_SERVER_ERROR,
-  headers:
-    isHttpErrorException(error) && Object.keys(error.headers).length >= 1
+  headers: {
+    ...defaultHeaders,
+    ...(isHttpErrorException(error) && Object.keys(error.headers).length >= 1
```
Comment (AlexHayton) [inline src/http/http.error.handler.ts:23]: Couldn't we just do `isHttpErrorException(error) ? error.headers : {}` now?

**Review by AlexHayton — CHANGES_REQUESTED** [review]
Looking pretty solid, but I think we need to add a test case to demonstrate that we're dealing with "an unhandled exception that has default headers defined" correctly.

Also a minor tidy up suggestion...

File: src/http/http.error.handler.ts (hunk 1 above)
Comment (bashleigh) [reply src/http/http.error.handler.ts:23]: then you don't get your defaults?

Comment (AlexHayton) [reply src/http/http.error.handler.ts:23]: I mean the entire block would look like this:
```
headers: {
    ...defaultHeaders,
    ...isHttpErrorException(error) ? error.headers : {}
},
```

Comment (AlexHayton) [reply src/http/http.error.handler.ts:23]: In the above case the two get merged, so all those lovely CORS headers should still be there.

Comment (bashleigh) [reply src/http/http.error.handler.ts:23]: I think originally I wanted to return undefined so headers didn't exist if there were none but tbh, we could return them anyway. Especially now I'm returning an empty object so headers are always set as empty array regardless

Comment (bashleigh) [reply src/http/http.error.handler.ts]: added test

**Pull request merged**